src/Components/Components/src/Routing/Router.cs

```diff
@@ -277,14 +279,19 @@ private void OnLocationChanged(object sender, LocationChangedEventArgs args)
     }
 
     Task IHandleAfterRender.OnAfterRenderAsync()
+    {
+        return EnableNavigationInterceptionThenScrollToLocationHash();
+    }
+
+    private async Task EnableNavigationInterceptionThenScrollToLocationHash()
     {
         if (!_navigationInterceptionEnabled)
         {
             _navigationInterceptionEnabled = true;
-            return NavigationInterception.EnableNavigationInterceptionAsync();
+            await NavigationInterception.EnableNavigationInterceptionAsync();
         }
 
-        return Task.CompletedTask;
+        await ScrollToLocationHash.ScrollToLocationHash(_locationAbsolute);
```

Comment:
Does this mean that every time the `<Router>` renders, it will re-scroll to the current URL hash? If so that seems like incorrect behavior because it could result in the user's scroll position changing unexpectedly, unrelated to navigation.

To clarify, as far as I'm aware, the router could re-render at any time - not necessarily just because some navigation happened. It would happen any time time component containing the `<Router>` re-renders, which can be for arbitrary application-specific reasons.

If this has been accounted for somehow then sorry for the confusion, but please clarify if so!

Reply:
> Does this mean that every time the <Router> renders, it will re-scroll to the current URL hash?

Yes, it does.

Thank you! I didn't know that `Router` could re-render at any time.

I just found a better way of achieving the same without changing `Router.OnAfterAsync`. The challenge was to catch a moment when the page is rendered. [Here](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Web.JS/src/Rendering/Renderer.ts#L90) I will add scrolling to the element.